PharmacyApi/Controllers/PharmacyController.cs

```diff
@@ -24,29 +24,34 @@ public class PharmacyController : ControllerBase
 
     #endregion
 
-    [HttpGet]
+    [HttpPost]
     [Route("all")]
-    public IActionResult GetAll() => Ok(_pharmacyService.GetAll());
-    
-    [HttpGet]
-    [Route("{id}")]
-    public async Task<IActionResult> GetById(int id)
+    [Route("{id?}")]
+    public async Task<IActionResult> GetPharmacy(int? id = null)
     {
-        var pharmacy = await _pharmacyService.GetById(id);
-
-        return pharmacy is not null
-            ? Ok(pharmacy)
-            : StatusCode(404, $"No pharmacy found with id {id}");
+        if (id is null)
+        {
+            var pharmacyListResult = await _pharmacyService.GetPharmacyListAsync();
+            return pharmacyListResult.Success 
+                ? Ok(pharmacyListResult.Result) 
+                : NotFound(pharmacyListResult.ErrorMessage);
+        }
+
+        var pharmacyResult = await _pharmacyService.GetByIdAsync(id.Value);
+        return pharmacyResult.Success 
+            ? Ok(pharmacyResult.Result) 
+            : NotFound(pharmacyResult.ErrorMessage);
     }
+    
 
     [HttpPost]
-    [Route("update/{id}")]
-    public async Task<IActionResult> UpdateById(int id, Pharmacy pharmacy)
+    [Route("update")]
+    public async Task<IActionResult> UpdateById(Pharmacy pharmacy)
     {
-        var updatedPharmacy = await _pharmacyService.UpdateById(id, pharmacy);
+        var updatedPharmacyResult = await _pharmacyService.UpdateByIdAsync(pharmacy);
 
-        return updatedPharmacy is not null
-            ? Ok(updatedPharmacy)
-            : StatusCode(404, $"No pharmacy found with id {id}");
+        return updatedPharmacyResult.Success
+            ? Ok(updatedPharmacyResult.Result)
+            : NotFound(updatedPharmacyResult.ErrorMessage);
```

Comment:
Better, but NotFound means "404 - Page not found" not "Pharmacy Not Found..."
Another reason why i dislike RestFulness... twists the meaning up...
When you cant find an entity - that should be a different error code...

Reply:
How could I make this more elegant?